### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by coot**
This is based on not yet merged: https://github.com/input-output-hk/cardano-node/pull/1308

### CODE REVIEW

File: cardano-db-sync/src/Cardano/DbSync/Era.hs [hunk 1]
```diff
@@ -101,7 +101,8 @@ mkShelleyTopLevelConfig sgc =
     -- Can use Nothing for the last field because that field will be dropped
     -- by 'pInfoConfig' anyway.
     shelleyInfo :: ProtocolInfo IO ShelleyBlock
-    shelleyInfo = protocolInfoShelley sgc (ProtVer 0 0) Nothing
+    -- TODO: IS @'maxMajorPV' = 0@ THIS RIGHT?
+    shelleyInfo = protocolInfoShelley sgc 0 (ProtVer 0 0) Nothing
```
Comment (coot) [inline cardano-db-sync/src/Cardano/DbSync/Era.hs:105]: @erikd I don't know what value should be passed here, probably something else than `0`.

Comment (erikd) [reply cardano-db-sync/src/Cardano/DbSync/Era.hs:105]: Neither do I. Complete mystery to me.

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs [hunk 2]
```diff
@@ -280,16 +278,26 @@ genesisUtxOs sg =
...
-      Shelley.Addr _networkId (Shelley.ScriptHashObj (Shelley.ScriptHash h)) _sref -> pseudoTxIn h
+      Shelley.Addr _networkId (Shelley.KeyHashObj    (Shelley.KeyHash    h)) _sref -> pseudoTxInAddr h
+      Shelley.Addr _networkId (Shelley.ScriptHashObj (Shelley.ScriptHash h)) _sref -> pseudoTxInHash h
       Shelley.AddrBootstrap byronAddr ->
         panic $ "Unsupported Byron address in the genesis UTxO: " <> show byronAddr
   where
-    pseudoTxIn :: Crypto.Hash (HASH TPraosStandardCrypto) a -> ShelleyTxIn
-    pseudoTxIn h = Shelley.TxIn (pseudoTxId h) 0
+    pseudoTxInAddr :: Crypto.Hash (ADDRHASH TPraosStandardCrypto) a -> ShelleyTxIn
+    pseudoTxInAddr h = Shelley.TxIn (pseudoTxIdAddr h) 0
```
Comment (coot) [inline cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:287]: I couldn't find a way to combine `pseudoTxInAddr` and `pseudoTxInHash` as a single polymorphic function.

**Comment by coot** [comment]
I wasn't able to update `cabal.project` and `stack.yaml` files because of: https://github.com/input-output-hk/cardano-repo-tool/issues/18

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs (hunk 2 above)
Comment (erikd) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:287]: That functionality has moved to `ledger-specs` and I should be using that.

File: cardano-db-sync/src/Cardano/DbSync/Era.hs (hunk 1 above)
Comment (coot) [reply cardano-db-sync/src/Cardano/DbSync/Era.hs:105]: @mrBliss @nc6 any idea what should be put here?

Comment (mrBliss) [reply cardano-db-sync/src/Cardano/DbSync/Era.hs:105]:
```suggestion
    shelleyInfo = protocolInfoShelley sgc 2 (ProtVer 2 0) Nothing
```
I believe that the major protocol version for Shelley will be 2, there was a discussion about it on Slack (OBFT is 1.0.0).

The `maxMajorPV` is the maximum major protocol version that is allowed by the ledger rules. I don't know what we should use, but it should at least be >= the major version of Shelley. Maybe @JaredCorduan knows a good value for `maxMajorPV`?

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs [hunk 3]
```diff
@@ -249,13 +247,13 @@ unCoin (Shelley.Coin c) = fromIntegral c
 -- it as milliseconds.
 configSlotDuration :: ShelleyGenesis TPraosStandardCrypto -> Word64
 configSlotDuration sg =
-  floor $ 1000 * Shelley.getSlotLength (Shelley.sgSlotLength sg)
+  floor $ 1000 * (Shelley.sgSlotLength sg)
```
Comment (mrBliss) [inline cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:250]: `Use Ouroboros.Consensus.BlockchainTime.WallClock.slotLengthToMillisec`

Comment (coot) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:250]: ok, thanks

Comment (mrBliss) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:250]: There's also `mkSlotLength :: NominalDiffTime -> SlotLength` in that same module.

Comment (coot) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:250]: done

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs (hunk 2 above)
Comment (erikd) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs:291]: Will do that in an upcoming PR.

**Comment by coot** [comment]
Thanks @erikd for taking care about this PR.

**Pull request merged**